src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -19,22 +23,158 @@ class WP_HTML_Processor extends WP_HTML_Tag_Processor {
+		 */
+		if ( 'HTML' === $tag_name && 'opener' === $tag_type ) {
+			goto ignored;
+		}
+
+		/*
+		 * > A start tag whose tag name is one of: "base", "basefont", "bgsound",
+		 * > "link", "meta", "noframes", "script", "style", "template", "title"
+		 *
+		 * > An end tag whose tag name is "template"
+		 */
+		if (
+			'opener' === $tag_type && (
+				'BASE' === $tag_name ||
+				'BASEFONT' === $tag_name ||
+				'BGSOUND' === $tag_name ||
+				'LINK' === $tag_name ||
+				'META' === $tag_name ||
+				'NOFRAMES' === $tag_name ||
+				'SCRIPT' === $tag_name ||
+				'STYLE' === $tag_name ||
+				'TEMPLATE' === $tag_name ||
+				'TITLE' === $tag_name
+			) ||
+			(
+				'closer' === $tag_type &&
+				'TEMPLATE' === $tag_name
+			) )
+		{
+			parent::seek( 'current' );
+			$this->insertion_mode = 'in-head';
+			return $this->step();
+		}
+
+		/*
+		 * > A start tag whose tag name is "body"
+		 */
+		if ( 'opener' === $tag_type && 'BODY' === $tag_name ) {
+			goto ignored;
+		}
+
+		/*
+		 * > A start tag whose tag name is "frameset"
+		 */
+		if ( 'opener' === $tag_type && 'FRAMESET' === $tag_name ) {
+			return self::NOT_IMPLEMENTED_YET;
+		}
+
+		/*
+		 * > An end-of-file token
+		 *
+		 * Stop parsing.
+		 */
+
+		/*
+		 * > An end tag whose tag name is "body"
+		 * > An end tag whose tag name is "html"
+		 */
+		if ( 'closer' === $tag_type && ( 'BODY' === $tag_name || 'HTML' === $tag_name ) ) {
+			/*
+			 * > If the stack of open elements does not have a body element in scope, this is a parse error; ignore the token.
+			 *
+			 * @TODO: We didn't construct an open HTML or BODY tag, but we have to make a choice here based on that.
+			 *        Probably need to create these _or_ assume this will always transfer to "after body".
+			 */
+			$this->insertion_mode = 'after-body';
+			return true;
+		}
+
+		/*
+		 * > A start tag whose tag name is one of: "address", "article", "aside", "blockquote", "center",
+		 * > "details", "dialog", "dir", "div", "dl", "fieldset", "figcaption", "figure", "footer",
+		 * > "header", "hgroup", "main", "menu", "nav", "ol", "p", "search", "section", "summary", "ul"
+		 */
+		if (
+			'opener' === $tag_type && (
+				'ADDRESS' === $tag_name ||
+				'ARTICLE' === $tag_name ||
+				'ASIDE' === $tag_name ||
+				'BLOCKQUOTE' === $tag_name ||
+				'CENTER' === $tag_name ||
+				'DETAILS' === $tag_name ||
+				'DIALOG' === $tag_name ||
+				'DIR' === $tag_name ||
+				'DIV' === $tag_name ||
+				'DL' === $tag_name ||
+				'FIELDSET' === $tag_name ||
+				'FIGCAPTION' === $tag_name ||
+				'FIGURE' === $tag_name ||
+				'FOOTER' === $tag_name ||
+				'HEADER' === $tag_name ||
+				'HGROUP' === $tag_name ||
+				'MAIN' === $tag_name ||
+				'MENU' === $tag_name ||
+				'NAV' === $tag_name ||
+				'OL' === $tag_name ||
+				'P' === $tag_name ||
+				'SEARCH' === $tag_name ||
+				'SECTION' === $tag_name ||
+				'SUMMARY' === $tag_name ||
+				'UL' === $tag_name
+			)
+		) {
+			if ( $this->has_in_scope( 'P', 'BUTTON' ) ) {
```

Comment:
I originally used the name `is_element_in_button_scope` because it's the exact term used in the documentation. They use a few similar ones like at least `is_element_in_scope` and `is_element_in_list_item_scope` and I found that not using the original terms makes side-by-side debugging very confusing.